python312Packages.phx-class-registry: 4.1.0 -> 5.0.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.phx-class-registry is: Factory and registry pattern for Python classes

meta.homepage for python312Packages.phx-class-registry is: https://class-registry.readthedocs.io/en/latest/




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 5.0.0 in filename of file in /nix/store/j7yv91wp3qz3hnsjsjvxz0z0hi39cx0m-python3.12-class-registry-5.0.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
4 total rebuild path(s)

4 package rebuild(s)

First fifty rebuilds by attrpath

backblaze-b2
python311Packages.phx-class-registry
python312Packages.phx-class-registry
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/j7yv91wp3qz3hnsjsjvxz0z0hi39cx0m-python3.12-class-registry-5.0.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.phx-class-registry https://github.com/r-ryantm/nixpkgs/archive/a9c1dd2ac8bbe6b11ff0bfd4e75909ac9dd46a06.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/a9c1dd2ac8bbe6b11ff0bfd4e75909ac9dd46a06#python312Packages.phx-class-registry
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/j7yv91wp3qz3hnsjsjvxz0z0hi39cx0m-python3.12-class-registry-5.0.0
ls -la /nix/store/j7yv91wp3qz3hnsjsjvxz0z0hi39cx0m-python3.12-class-registry-5.0.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.phx-class-registry</li>
    <li>python311Packages.phx-class-registry.dist</li>
    <li>python312Packages.phx-class-registry</li>
    <li>python312Packages.phx-class-registry.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @hrdinka @tomhoule for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
python312Packages.phx-class-registry: 4.1.0 -> 5.0.0

natsukium commented:
Result of `nixpkgs-review pr 340420` at a9c1dd2a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/c3054ae3c26862be3554a9b6e2919613">backblaze-b2</a></li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python311Packages.phx-class-registry</li>
    <li>python312Packages.phx-class-registry</li>
  </ul>
</details>

natsukium committed:
python312Packages.phx-class-registry: 5.0.0 -> 5.1.1

Diff: https://github.com/todofixthis/class-registry/compare/refs/tags/5.0.0...5.1.1

phaer commented:
Already on 5.1.1 in nixos-unstable
